server/routes/messages.js

```diff
@@ -6,6 +6,33 @@ const mustBeAuthenticated = require("../middleware");
 //Middleware to ensure authentication
 messagesRouter.use(mustBeAuthenticated);
 
+//Route to create a new group (POST /groups)
+messagesRouter.post('/groups', async (req, res) => {
+    try {
+      //Extract necessary data from request body
+      const { groupName, selectedUsers } = req.body;
+
+      if (!groupName || !selectedUsers || Array.isArray(selectedUsers)) {
+        return res
+          .status(400)
+          .send({ msg: "Invalid request. Missing required data" });
+      }
+
+      //Create a new group
+      const group = new Chat({ name: groupName, members: selectedUsers });
```

Comment:
change name from `group` to `chat`

Reply:
@DanielC34, where are you getting the `members` and `name` arguments from? chat schema doesn't have such fields

Reply:
Chat schema only has `type` and `participants`

Reply:
So what should line 22 contain? like inside the curly braces of new Chat

Reply:
```suggestion
      const group = new Chat({ type:"group", participants: selectedUsers, name: groupName });
```
@DanielC34  You will need to introduce a new field called name with the schema. the name will store the group name for group change and will be black for private chats

Reply:
Wait did you not say i should change selectedUsers to user._id earlier?

Reply:
No, i meant the variable name should change to `userIds` or `participantIds` or `recipientIds`

Reply:
so for name in the chat model file, name should be a required string right? am i missing anything on that part?

Reply:
When you create the name field, is should not be required on the schema level,

Reply:
so it should be like this: name: { type: String },

Reply:
yes

Reply:
Now after making this change, where do i look into next relating to this? I just changed all the variable names and did as you said

Reply:
Weird how when i try to load up the add group card its not bringing up the users after i made these changes